pkg/controller/sparkapplication/controller.go

```diff
@@ -150,6 +158,34 @@ func newSparkApplicationController(
 	return controller
 }
 
+func getEventResourceHandlerFuncs(handlerFuncs cache.ResourceEventHandlerFuncs, namespaceFilter *regexp.Regexp) cache.ResourceEventHandler {
+	var resourceEventHandlerFuncs cache.ResourceEventHandler
+
+	if namespaceFilter == nil {
```

Comment:
You can simplify the logic of this function:

```
if namespaceFilter == nil {
    return handlerFuncs
}

return cache.FilteringResourceEventHandler{
    FilterFunc: filterSparkAndPodEventsByNamespace(*namespaceFilter),
    Handler:    handlerFuncs,
}
```

Reply:
Done.